.github/workflows/typescript-publish.yml

```diff
@@ -0,0 +1,55 @@
+#
+# Copyright (c) 2021 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
+name: typescript-publish-next
+
+on:
+  workflow_dispatch:
+  push:
+    branches: 
+      - main
+      - 7.**.x
+    paths:
+    - 'tests/e2e/**'
+    - '.github/workflows/typescript-publish.yml'
+
+jobs:
+  publish:
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: actions/setup-node@v1
+      with:
+        node-version: '12'
+        registry-url: 'https://registry.npmjs.org'
+        scope: '@eclipse-che'
+    - name: Clone source code
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: publish
+      env:
+        NODE_AUTH_TOKEN: ${{secrets.CHE_NPM_AUTH_TOKEN}}
+      run: |
+        if [[ ${GITHUB_REF##*/} == "7."**".x" ]]; then
+          echo "[INFO] using ${GITHUB_REF##*/} tag"
+          DIST_TAG=${GITHUB_REF##*/}
+        else
+          echo "[INFO] using "next" tag"
+          DIST_TAG=next
+        fi
+
+        cd tests/e2e
+        npm ci && npm run tsc
```

Comment:
So, will we be doing the `npm audit --fix` and pushing package-lock.json manually?

Reply:
I would vote for package-lock.json being updated "manually" by test developers and anything in CI would use `npm  ci`.
The "natural" update of package-lock.json will happen by merging devs PRs (Devs will naturally do "npm install" locally, which should update package-lock.json and this one should be commited&pushed together with their PR).

As for `npm audit` - From what I read `npm audit fix` will update just within the ranges defined in package.json. To properly udpate deps to resolve the vulnerabilities we would need to run `npm audit fix --force` which updates package.json as well.

Reply:
@mkuznyetsov - Please execute `npm audit fix --force` and update the package.json, package.lock.json. So we are good to merge the pr.
cc: @dmytro-ndp

Reply:
@mkuznets - On `npm audit fix --force` the mocha version is updating from `7.1.1` to `9.1.3`,  Where opts files are [deprecated](https://main-jenkins-csb-crwqe.apps.ocp4.prod.psi.redhat.com/blue/organizations/jenkins/Che%2Fe2e%2Fminikube%2Fpr-check%2Fche-tests-pr-check/detail/che-tests-pr-check/287/pipeline/). IMO its better to have a separate pr for these changes.